File: app/src/main/java/com/maciejkozlowski/databases/objectbox/CitiesLoaderBox.java
package com.maciejkozlowski.databases.objectbox;


import android.content.Context;

import com.maciejkozlowski.databases.BaseLoader;
import com.maciejkozlowski.databases.Timings;
import com.maciejkozlowski.databases.results.ResultSet;

import java.util.List;

import io.objectbox.Box;

/**
 * Created by Maciej Kozłowski on 01.05.17.
 */
public class CitiesLoaderBox extends BaseLoader<CityBox> {

    public static final String TAG = "box";

    public void execute(Context context, ResultSet resultSet, Box<CityBox> boxStore, int size) {
        List<CityBox> cities = readFromFile(context, Companion.getCITIES_CSV(), size);

        getLogger().start();
        boxStore.put(cities);
        getLogger().logTime(resultSet.getCreating(), Companion.getINSERT_CITIES(), size);

        getLogger().start();
        List<CityBox> citiesBox = boxStore.getAll();
        getLogger().logTime(resultSet.getReading(), Companion.getREAD_CITIES(), size);


        getLogger().start();
        for (int i = 0; i < citiesBox.size(); i++) {
            citiesBox.get(i).setName(String.valueOf(i));
        }
        boxStore.put(citiesBox);
        getLogger().logTime(resultSet.getUpdating(), Companion.getUPDATE_CITIES(), size);

        getLogger().start();
        boxStore.removeAll();
        getLogger().logTime(resultSet.getDeleting(), Companion.getDELETE_CITIES(), size);
    }

    @Override
    protected CityBox create(Long id, String name, Double latitude, Double longitude) {
        return new CityBox(id, name, latitude, longitude);
    }

    @Override
    protected Timings createTimingLogger() {
        return new Timings(TAG);
    }
}

File: app/src/main/java/com/maciejkozlowski/databases/objectbox/CitiesLoaderBox.kt
package com.maciejkozlowski.databases.objectbox


import android.content.Context

import com.maciejkozlowski.databases.BaseLoader
import com.maciejkozlowski.databases.Timings
import com.maciejkozlowski.databases.results.ResultSet

import io.objectbox.Box

/**
 * Created by Maciej Kozłowski on 01.05.17.
 */
class CitiesLoaderBox : BaseLoader<CityBox>() {

    fun execute(context: Context, resultSet: ResultSet, boxStore: Box<CityBox>, size: Int) {
        val cities = readFromFile(context, CITIES_CSV, size)

        logger.start()
        boxStore.put(cities)
        logger.logTime(resultSet.creating, INSERT_CITIES, size)

        logger.start()
        val citiesBox = boxStore.all
        logger.logTime(resultSet.reading, READ_CITIES, size)


        logger.start()
        for (i in citiesBox.indices) {
            citiesBox[i].name = i.toString()
        }
        boxStore.put(citiesBox)
        logger.logTime(resultSet.updating, UPDATE_CITIES, size)

        logger.start()
        boxStore.removeAll()
        logger.logTime(resultSet.deleting, DELETE_CITIES, size)
    }

    override fun create(id: Long?, name: String, latitude: Double?, longitude: Double?): CityBox {
        return CityBox(id!!, name, latitude!!, longitude!!)
    }

    override fun createTimingLogger(): Timings {
        return Timings(TAG)
    }

    companion object {

        const val TAG = "box"
    }
}
